src/panel/poi/PoiPanel.jsx

```diff
@@ -57,15 +48,9 @@ export default class PoiPanel extends React.Component {
     // Load poi or pois
     !this.props.pois ? this.loadPoi() : this.loadPois();
 
-    this.storeAddHandler = listen('poi_added_to_favs', poi => {
-      if (poi === this.state.fullPoi) {
-        this.setState({ isPoiInFavorite: true });
-      }
-    });
-
-    this.storeRemoveHandler = listen('poi_removed_from_favs', poi => {
+    this.storeAddRemoveHandler = listen('poi_favorite_state_changed', (poi, isPoiInFavorite) => {
```

Comment:
What about a function name like `storePoiChangeHandler`, somethink like this ?

Reply:
Not sure I understand, can you write some code?

Reply:
I just suggest to rename the handler: `storeAddRemoveHandler` -> `storePoiChangeHandler`. To reflect its event name `poi_favorite_state_changed` more closely. Wdyt ?

Reply:
Alright. Thanks for the explanation!

Reply:
Done :)